src/borg/archive.py

```diff
@@ -365,6 +373,16 @@ def write_checkpoint(self):
         del self.manifest.archives[self.checkpoint_name]
         self.cache.chunk_decref(self.id, self.stats)
 
+    def get_summary(self):
+        summary = namedtuple('Summary', ['name', 'fingerprint', 'comment',
+        'hostname', 'username', 'time_start', 'time_end', 'duration', 'command_line'])
+        return summary(*self.summarize())._asdict()
+
+    def summarize(self):
+        return [self.name, self.fpr, self.metadata.get('comment', ''),
+        self.metadata.hostname, self.metadata.username, format_time(to_localtime(self.ts)),
+        format_time(to_localtime(self.ts_end)), self.duration_from_meta, self.metadata.cmdline]
```

Comment:
I think it would be more clear if you perhaps notated this more directly, eg. the namedtuple isn't really used.

```py
def get_summary(self):
    return {
        'name': self.name,
        'fingerprint': self.fpr,
        # ... rest like you did it
    }
```

What do you think?